### ISSUE TITLE
Detaching attributions from computation graph

### ISSUE THREAD

**Issue by aarzchan** [pytorch/captum#793]
Hi! I'm interested in regularizing the model via some attribution-based loss, as described in a [previous post](https://github.com/pytorch/captum/issues/761#issue-996495482). As a baseline, I would like to train a model without such regularization (i.e., using only task loss), but using the attributions to compute some evaluation metric.

The desired workflow for this baseline is as follows:

1. Compute the gradient-based attributions (e.g., using IG) for the model, but without keeping the gradients used to obtain the attributions. That is, all I want is the final attribution vector, before starting with an empty computation graph for Step 2.
2. Perform a forward pass through the model, use the task labels to compute the loss, then backprop this loss only via gradients computed in Step 2. In other words, when I perform the backward pass here, I don't want there to be any connection to the computation in Step 1.
3. Use the attributions obtained from Step 1 to compute the evaluation metric.

I've tried doing `detach` and `torch.no_grad` in Step 1 as well as removing Step 3, but the train loss in Step 2 is somehow still being affected by the attribution computation.

I'd appreciate any advice on how to resolve this. Thanks!

**Comment by aarzchan** [comment]
@NarineK 

That's a good point! My model is using dropout, so let me check what happens when I turn off dropout in Step 1.

However, in eval mode, why does GradientShap behave differently from the other attribution algorithms?

**Comment by craymichael** [comment]
This issue appears to have been answered in the discussion above, so I will close it for now. Please feel free to reopen or follow up if anything is still unresolved.
